### PR TITLE
Add category for track to forum

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Reduce nesting for Track::Create
- Extract seeding a track to command
- Add tests to verify creating in correct position
- Create track category on forum

### CODE REVIEW

File: app/commands/forum/create_track_category.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
+class Forum::CreateTrackCategory
```
Comment (ErikSchierboom) [inline app/commands/forum/create_track_category.rb]: Let me know if you have a better suggestion for the name. 
I considered `SyncTrack`, `AddTrack` and `CreateTrack`.

Comment (iHiD) [reply app/commands/forum/create_track_category.rb]: Instinctively I'd go `Track::CreateForumCategory` as Forum isn't really a primative in our world.

**Comment by ErikSchierboom** [comment]
Updated to use the discourse client from exercism_config.

File: app/commands/user/set_discourse_trust_level.rb [hunk 2]
```diff
@@ -6,17 +6,13 @@ class User::SetDiscourseTrustLevel
...
   end
 
-  def client
-    DiscourseApi::Client.new("https://forum.exercism.org").tap do |client|
-      client.api_key = Exercism.secrets.discourse_api_key
-      client.api_username = "system"
-    end
-  end
+  memoize
+  def client = Exercism.discourse_client
```
Comment (iHiD) [inline app/commands/user/set_discourse_trust_level.rb:17]: I don't think you need that here?

**Review by iHiD — APPROVED** [review]

**Review by iHiD — APPROVED** [review]

**Pull request merged**